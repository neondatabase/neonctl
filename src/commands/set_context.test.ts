import { tmpdir } from 'node:os';
import { join } from 'node:path';
import { rmSync } from 'node:fs';
import { afterAll, describe } from '@jest/globals';
import { testCliCommand } from '../test_utils/test_cli_command';

const CONTEXT_FILE = join(tmpdir(), `neon_${Date.now()}`);

describe('set_context', () => {
  afterAll(() => {
    rmSync(CONTEXT_FILE);
  });

  describe('should set the context', () => {
    testCliCommand({
      name: 'set-context',
      args: [
        'set-context',
        '--project-id',
        'test',
        '--context-file',
        CONTEXT_FILE,
      ],
    });

    testCliCommand({
      name: 'list branches selecting project from the context',
      args: ['branches', 'list', '--context-file', CONTEXT_FILE],
      expected: {
        snapshot: true,
      },
    });
  });
});
