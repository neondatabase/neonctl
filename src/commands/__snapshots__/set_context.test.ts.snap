// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`set_context > should set the context > get project id overrides context set project > test 1`] = `
"id: project-id-123
name: project name 123
created_at: 2019-01-01T00:00:00Z
settings:
  allowed_ips:
    ips:
      - 192.168.1.1
    primary_branch_only: false
"
`;

exports[`set_context > should set the context > list branches selecting project from the context > test 1`] = `
"- id: br-main-branch-123456
  name: main
  default: true
  created_at: 2021-01-01T00:00:00.000Z
  updated_at: 2021-01-01T00:00:00.000Z
- id: br-sunny-branch-123456
  name: test_branch
  parent_id: br-parent-branch-123456
  created_at: 2021-01-01T00:00:00.000Z
  updated_at: 2021-01-01T00:00:00.000Z
- id: br-numbered-branch-123456
  name: "123"
  parent_id: br-parent-branch-123456
  created_at: 2021-01-01T00:00:00.000Z
  updated_at: 2021-01-01T00:00:00.000Z
"
`;

exports[`set_context > should set the context > set the branchId and projectId is from context > test 1`] = `
"- name: db1
  owner_name: user1
- name: db2
  owner_name: user2
- name: test_db
  owner_name: test_user
"
`;

exports[`set_context > should set the context > should not set branchId from context for non-context projectId > test 1`] = `""`;
